src/rpc/helpers.rs

```diff
@@ -0,0 +1,148 @@
+/*
```

Comment:
I would've left this code (together with the tests) in the client.rs, as you're only using it there. It also would have been a simpler change.

Reply:
Since both, ws and tungstenite client, are using these functions, I extracted it to the helpers. That way, I was able to remove duplicate code.